validphys2/src/validphys/config.py

```diff
@@ -995,13 +993,87 @@ def produce_matched_datasets_from_dataspecs(self, dataspecs):
             # TODO: Should this have the same name?
             inner_spec_list = inres["dataspecs"] = []
             for ispec, spec in enumerate(dataspecs):
-                # Passing spec by referene
+                # Passing spec by reference
                 d = ChainMap({"dataset_input": all_names[ispec][k]}, spec)
                 inner_spec_list.append(d)
             res.append(inres)
         res.sort(key=lambda x: (x["process"], x["dataset_name"]))
         return res
 
+#    def produce_matched_excluded_datasets_by_name(self, dataspecs):
+#        """
+#        Return excluded datasets, each tagged with the more_info from the dataspecs they came from. Set up to work with plot_fancy.
+#        """
+#        dinputs_a = dataspecs[0]["dataset_inputs"]
+#        dinputs_b = dataspecs[1]["dataset_inputs"]
+#        # some fancy logic
+#        more_info = {
+#                "pdfs": [i["pdf"] for i in dataspecs],
+#                "theoryid": dataspecs[0]["theoryid"],
+#                "fit": dataspecs[0]["fit"] # The fancy logic will put here the _right_ fit since this is used for the cuts
+#        }
+#        return [{"dataset_input": i,
+#                 "dataset_name": i.name, 
+#                 **more_info} for i in dinputs_b[1:3]]
+
+    def produce_matched_excluded_datasets_by_name(self, dataspecs):
+        """Return excluded datasets, each tagged with the more_info from the dataspecs they came from."""
+        """
+        Like produce_matched_datasets_from_dataspecs, but for excluded datasets from a fit comparison.
+        Return excluded datasets, each tagged with the more_info from the dataspecs they came from. Set up to work with plot_fancy.
+        """
+
+        self._check_dataspecs_type(dataspecs)
+        loader = Loader()
+
+        # (proc, ds) -> list of (dsin, spec)
+        all_sets = {}
+
+        for spec in dataspecs:
+            with self.set_context(ns=self._curr_ns.new_child(spec)):
+                _, data_input = self.parse_from_(None, "data_input", write=False)
+                for dsin in data_input:
+                    cd = self.produce_commondata(dataset_input=dsin)
+                    proc = get_info(cd).nnpdf31_process
+                    ds = dsin.name
+                    key = (proc, ds)
+
+                    # add (dsin, spec) to the set containing (proc, ds)
+                    if key not in all_sets:
+                        all_sets[key] = []
+                    all_sets[key].append((dsin, spec))
```

Comment:
What are you exactly doing here? If you are just checking the `dataset_input` from the `dataspecs` by name you should not need a lot of this.

Note that you are trying to reproduce `produce_matched_datasets_from_dataspecs` but what you _really_ want is `produce_matched_excluded_dataset_inputs_by_name`. And what you are really doing, this can be simplified quite a bit.